### PR TITLE
feat(csp): use cheerio parser

### PR DESCRIPTION
**Pull request by vejja**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This PR improves the way we parse HTML elements to insert hashes and nonces in CSP policies.
We rely on Cheerio instead of regexes.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)

Tests pass

### CODE REVIEW

File: src/defaultConfig.ts [hunk 1]
```diff
@@ -81,6 +81,7 @@ export const defaultSecurityConfig = (serverlUrl: string): ModuleOptions => ({
     exclude: [/node_modules/, /\.git/]
   },
   ssg: {
-    hashScripts: true
+    hashScripts: true,
+    hashStyles: false
```
Comment (Baroshem) [inline src/defaultConfig.ts:85]: suggestion. I think that hash styles will be introduced in another PR. Could you please revert it so that we can only ship the cheerio addition with this PR? :) 

The hash styles I was hoping to release in the next RC version.

Comment (vejja) [reply src/defaultConfig.ts]: Done!

**Comment by vejja** [comment]
Let me know if you want me to rebase for clarity

**Review by Baroshem — APPROVED** [review]
